scripts/create-kubestellar-demo-env.sh

```diff
@@ -139,9 +104,83 @@ checking_cluster() {
 
     done
 }
+##########################################
+
+echo -e "\nStarting environment clean up..."
+echo -e "Starting cluster clean up..."
+
+cluster_clean_up "kind delete cluster --name kubeflex"
+cluster_clean_up "kind delete cluster --name cluster1"
+cluster_clean_up "kind delete cluster --name cluster2"
+echo -e "Cluster space clean up has been completed"
+
+echo -e "\nStarting context clean up..."
+
+context_clean_up
+echo "Context space clean up completed"
+
+echo -e "\nStarting the process to install KubeStellar core: kind-kubeflex..."
+
+clusters=(cluster1 cluster2)
+
+for cluster in "${clusters[@]}"; do
+   (
+     echo -e "Creating cluster ${cluster}..."
+     kind create cluster --name "${cluster}" >/dev/null 2>&1 &&
+     kubectl config rename-context "kind-${cluster}" "${cluster}" >/dev/null 2>&1
+     echo -e "${cluster} creation and context setup complete"
+   ) &
+done
+
+wait 
+
+echo -e "Creating KubeFlex cluster with SSL Passthrough"
+curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v${KUBESTELLAR_VERSION}/scripts/create-kind-cluster-with-SSL-passthrough.sh | bash -s -- --name kubeflex --port 9443 
+# >/dev/null 2>&1
+
+echo -e "Completed KubeFlex cluster with SSL Passthrough"
+
+(docker pull ghcr.io/loft-sh/vcluster:0.16.4 && kind load docker-image ghcr.io/loft-sh/vcluster:0.16.4 --name kubeflex) & 
+(docker pull rancher/k3s:v1.27.2-k3s1 && kind load docker-image rancher/k3s:v1.27.2-k3s1 --name kubeflex) &
+(docker pull quay.io/open-cluster-management/registration-operator:v0.13.2 && kind load docker-image quay.io/open-cluster-management/registration-operator:v0.13.2 --name kubeflex) & 
+
+# curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v${KUBESTELLAR_VERSION}/scripts/create-kind-cluster-with-SSL-passthrough.sh | bash -s -- --name kubeflex --port 9443
+
+
+helm upgrade --install ks-core oci://ghcr.io/kubestellar/kubestellar/core-chart \
+    --version $KUBESTELLAR_VERSION \
+    --set-json='ITSes=[{"name":"its1"}]' \
+    --set-json='WDSes=[{"name":"wds1"}]' \
+    --set-json='verbosity.default=5'
+
+kubectl config delete-context its1 || true
+kflex ctx its1
+kubectl config delete-context wds1 || true
+kflex ctx wds1
+
+# switch back to its1 context to crete 2 remote clusters, add OCM agents to them, and register them back to the KubeStellar core
+kflex ctx its1
```

Comment:
The commands that follow do not care what the kubeconfig current context is, they all explicitly specify a context (which is a good thing, I think we have agreed). So I think this command can simply be removed. I have tested a modified version of this script that omits this command, and it worked.

Reply:
ive tested without this and it does not allow the OCM bits to work properly - leaving these commands in

Reply:
You have re-ordered the first two context establishment commands and eliminated the third (as it would clearly be redundant).